src/Optical/OpticalSystem.jl

```diff
@@ -180,7 +181,7 @@ function trace(
             # this will almost always not apply as the detector will be in air, but it's possible that the detector is
             # not in air, in which case this is necessary
             if !isair(m)
-                mat = glassforid(m)::OpticSim.GlassCat.Glass
+                mat::Glass = glassforid(m)
```

Comment:
Is `glassforid(m)::OpticSim.GlassCat.Glass` equivalent to `convert(OpticSim.GlassCat.Glass, glassforid(m)`, or is it more of a type assertion?

Either way, in this case, am I right to say that the type declaration `mat::Glass` is preferable?

Reply:
```
mat = glassforid(m)::OpticSim.GlassCat.Glass
```
says that mat will be assigned something of type Glass but does not fix the type of mat.
```
mat::Glass = glassforid(m)
```
fixes the type of mat. The latter allows the compiler to make more assumptions about mat, since it's type is not fixed and cannot change anywhere else in the program. Probably better to do it this way in general.